### PR TITLE
Fix inconsistent builds by sorting dir_entries

### PR DESCRIPTION
**Pull request by jul-sh**
Thanks for bearing with me, reopening #601 with fixes to ensure it compiles properly.

### CODE REVIEW

File: linalg/build.rs [hunk 1]
```diff
@@ -153,8 +153,13 @@ fn preprocess_files(
 ) -> Vec<path::PathBuf> {
     let out_dir = path::PathBuf::from(var("OUT_DIR"));
     let mut files = vec![];
-    for f in input.as_ref().read_dir().unwrap() {
-        let f = f.unwrap();
+    let dir_entries = {
+        let mut dir_entries: Vec<fs::DirEntry> =
+            input.as_ref().read_dir().unwrap().map(|f| f.unwrap()).collect();
+        dir_entries.sort_by(|a, b| a.file_name().cmp(&b.file_name()));
```
Comment (jul-sh) [inline linalg/build.rs:159]: Upon reflection this should probably sort based on the full path, will add commit to do that

**Comment by kali** [comment]
Hey, want to rebase and check if CI tweak is working ? :)

**Comment by jul-sh** [comment]
done, running checks now :]

**Comment by kali** [comment]
great :) thanks.

**Pull request merged**